File: Lib/Telerik/Examples/DemoAppsHub/ListViewExample/CarsListVisualItem.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Telerik.WinControls.UI
Imports Telerik.WinControls.Layouts
Imports System.Drawing
Imports System.Windows.Forms

Namespace DemoAppsHub.ListViewExample
	Public Class CarsListVisualItem
		Inherits SimpleListViewVisualItem
		Private element1 As LightVisualElement
		Private element2 As LightVisualElement
		Private element3 As LightVisualElement
		Private element4 As LightVisualElement
        Private stackLayout As StackLayoutPanel

        Protected Overrides Sub CreateChildElements()
            MyBase.CreateChildElements()

            Me.stackLayout = New StackLayoutPanel()
            Me.stackLayout.EqualChildrenWidth = True
            Me.stackLayout.Margin = New Padding(180, 30, 0, 0)

            Me.element1 = New LightVisualElement()
            element1.TextAlignment = ContentAlignment.MiddleLeft
            element1.MinSize = New Size(170, 0)
            element1.NotifyParentOnMouseInput = True
            element1.ShouldHandleMouseInput = False
            Me.stackLayout.Children.Add(Me.element1)

            Me.element2 = New LightVisualElement()
            element2.TextAlignment = ContentAlignment.MiddleLeft
            element2.MinSize = New Size(170, 0)
            element2.NotifyParentOnMouseInput = True
            element2.ShouldHandleMouseInput = False
            Me.stackLayout.Children.Add(Me.element2)

            Me.element3 = New LightVisualElement()
            element3.TextAlignment = ContentAlignment.MiddleLeft
            element3.MinSize = New Size(170, 0)
            element3.NotifyParentOnMouseInput = True
            element3.ShouldHandleMouseInput = False
            Me.stackLayout.Children.Add(Me.element3)

            Me.element4 = New LightVisualElement()
            element4.TextAlignment = ContentAlignment.MiddleLeft
            element4.MinSize = New Size(170, 0)
            element4.NotifyParentOnMouseInput = True
            element4.ShouldHandleMouseInput = False
            Me.stackLayout.Children.Add(Me.element4)

            Me.Children.Add(Me.stackLayout)
        End Sub

		Private Function ContainsFeature(ByVal item As ListViewDataItem, ByVal feature As String) As Boolean
			Return Not item(feature) Is Nothing AndAlso Convert.ToInt32(item(feature)) <> 0
		End Function

		Protected Overrides Sub SynchronizeProperties()
			MyBase.SynchronizeProperties()

			Me.Text = "<html><span style=""color:#141718;font-size:14.5pt;"">" & String.Format("{0}", Me.Data("Make")) & " " & String.Format("{0}", Me.Data("Model")) & "</span>"

            Me.element1.Text = "<html><span style=""color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;"">" & _
                "Manufactured:<span style=""color:#13224D;font-family:Segoe UI;"">" & String.Format("{0}", Me.Data("CarYear")) & "</span>" & _
                "<br>Category:<span style=""color:#13224D;font-family:Segoe UI;"">" & String.Format("{0}", Me.Data("CategoryName")) & "</span></span>"

			If Me.ContainsFeature(Me.Data, "ABS") Then
				If Me.ContainsFeature(Me.Data, "ESR") Then
					Me.element2.Text = "<html><span style=""color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;"">" & "ABS:" & ("<span style=""color:#13224D;font-family:Segoe UI;"">YES") & "</span>" & "<br>ESR:" & ("<span style=""color:#13224D;font-family:Segoe UI;"">YES") & "</span>" & "</span>"
				Else
					Me.element2.Text = "<html><span style=""color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;"">" & "ABS:" & ("<span style=""color:#13224D;font-family:Segoe UI;"">YES") & "</span>" & "<br>ESR:" & ("<span style=""color:#D71B0E;"">NO") & "</span>" & "</span>"
				End If
			Else
				If Me.ContainsFeature(Me.Data, "ESR") Then
					Me.element2.Text = "<html><span style=""color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;"">" & "ABS:" & ("<span style=""color:#D71B0E;"">NO") & "</span>" & "<br>ESR:" & ("<span style=""color:#13224D;font-family:Segoe UI;"">YES") & "</span>" & "</span>"
				Else
					Me.element2.Text = "<html><span style=""color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;"">" & "ABS:" & ("<span style=""color:#D71B0E;"">NO") & "</span>" & "<br>ESR:" & ("<span style=""color:#D71B0E;"">NO") & "</span>" & "</span>"
				End If
			End If

			If Me.ContainsFeature(Me.Data, "Mp3Player") Then
				If Me.ContainsFeature(Me.Data, "DVDPlayer") Then
					Me.element3.Text = "<html><span style=""color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;"">" & "MP3 Player:" & ("<span style=""color:#13224D;font-family:Segoe UI;"">YES") & "</span>" & "<br>DVD Player:" & ("<span style=""color:#13224D;font-family:Segoe UI;"">YES") & "</span>" & "</span>"
				Else
					Me.element3.Text = "<html><span style=""color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;"">" & "MP3 Player:" & ("<span style=""color:#13224D;font-family:Segoe UI;"">YES") & "</span>" & "<br>DVD Player:" & ("<span style=""color:#D71B0E;"">NO") & "</span>" & "</span>"
				End If
			Else
				If Me.ContainsFeature(Me.Data, "DVDPlayer") Then
					Me.element3.Text = "<html><span style=""color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;"">" & "MP3 Player:" & ("<span style=""color:#D71B0E;"">NO") & "</span>" & "<br>DVD Player:" & ("<span style=""color:#13224D;font-family:Segoe UI;"">YES") & "</span>" & "</span>"
				Else
					Me.element3.Text = "<html><span style=""color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;"">" & "MP3 Player:" & ("<span style=""color:#D71B0E;"">NO") & "</span>" & "<br>DVD Player:" & ("<span style=""color:#D71B0E;"">NO") & "</span>" & "</span>"
				End If
			End If

			If Me.ContainsFeature(Me.Data, "AirConditioner") Then
				If Me.ContainsFeature(Me.Data, "Navigation") Then
					Me.element4.Text = "<html><span style=""color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;"">" & "Air Conditioner:" & ("<span style=""color:#13224D;font-family:Segoe UI;"">YES") & "</span>" & "<br>Navigation:" & ("<span style=""color:#13224D;font-family:Segoe UI;"">YES") & "</span>" & "</span>"
				Else
					Me.element4.Text = "<html><span style=""color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;"">" & "Air Conditioner:" & ("<span style=""color:#13224D;font-family:Segoe UI;"">YES") & "</span>" & "<br>Navigation:" & ("<span style=""color:#D71B0E;"">NO") & "</span>" & "</span>"
				End If
			Else
				If Me.ContainsFeature(Me.Data, "Navigation") Then
					Me.element4.Text = "<html><span style=""color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;"">" & "Air Conditioner:" & ("<span style=""color:#D71B0E;"">NO") & "</span>" & "<br>Navigation:" & ("<span style=""color:#13224D;font-family:Segoe UI;"">YES") & "</span>" & "</span>"
				Else
					Me.element4.Text = "<html><span style=""color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;"">" & "Air Conditioner:" & ("<span style=""color:#D71B0E;"">NO") & "</span>" & "<br>Navigation:" & ("<span style=""color:#D71B0E;"">NO") & "</span>" & "</span>"
				End If
			End If

			Me.TextAlignment = ContentAlignment.TopLeft
		End Sub

		Protected Overrides ReadOnly Property ThemeEffectiveType() As Type
			Get
				Return GetType(SimpleListViewVisualItem)
			End Get
		End Property
	End Class
End Namespace

File: Lib/Telerik/Examples/DemoAppsHub/ListViewExample/CarsListVisualItem.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls.UI;
using Telerik.WinControls.Layouts;
using System.Drawing;
using System.Windows.Forms;

namespace DemoAppsHub.ListViewExample
{
    public class CarsListVisualItem : SimpleListViewVisualItem
    {
        private LightVisualElement element1;
        private LightVisualElement element2;
        private LightVisualElement element3;
        private LightVisualElement element4;
        private StackLayoutPanel layout;

        protected override void CreateChildElements()
        {
            base.CreateChildElements();

            this.layout = new StackLayoutPanel();
            this.layout.EqualChildrenWidth = true;
            this.layout.Margin = new Padding(180, 30, 0, 0);

            this.element1 = new LightVisualElement();
            element1.TextAlignment = ContentAlignment.MiddleLeft;
            element1.MinSize = new Size(170, 0);
            element1.NotifyParentOnMouseInput = true;
            element1.ShouldHandleMouseInput = false;
            this.layout.Children.Add(this.element1);

            this.element2 = new LightVisualElement();
            element2.TextAlignment = ContentAlignment.MiddleLeft;
            element2.MinSize = new Size(170, 0);
            element2.NotifyParentOnMouseInput = true;
            element2.ShouldHandleMouseInput = false;
            this.layout.Children.Add(this.element2);

            this.element3 = new LightVisualElement();
            element3.TextAlignment = ContentAlignment.MiddleLeft;
            element3.MinSize = new Size(170, 0);
            element3.NotifyParentOnMouseInput = true;
            element3.ShouldHandleMouseInput = false;
            this.layout.Children.Add(this.element3);

            this.element4 = new LightVisualElement();
            element4.TextAlignment = ContentAlignment.MiddleLeft;
            element4.MinSize = new Size(170, 0);
            element4.NotifyParentOnMouseInput = true;
            element4.ShouldHandleMouseInput = false;
            this.layout.Children.Add(this.element4);

            this.Children.Add(this.layout);
        }

        private bool ContainsFeature(ListViewDataItem item, string feature)
        {
            return item[feature] != null && Convert.ToInt32(item[feature]) != 0;
        }

        protected override void SynchronizeProperties()
        {
            base.SynchronizeProperties();

            this.Text = "<html><span style=\"color:#141718;font-size:14.5pt;\">" + this.Data["Make"] + " " + this.Data["Model"] + "</span>";

            this.element1.Text = "<html><span style=\"color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;\">" +
                "Manufactured:<span style=\"color:#13224D;font-family:Segoe UI;\">" + this.Data["CarYear"] + "</span>" +
                "<br>Category:<span style=\"color:#13224D;font-family:Segoe UI;\">" + this.Data["CategoryName"] + "</span></span>";

            this.element2.Text = "<html><span style=\"color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;\">" +
                "ABS:" + (this.ContainsFeature(this.Data, "ABS") ? "<span style=\"color:#13224D;font-family:Segoe UI;\">YES" : "<span style=\"color:#D71B0E;\">NO") + "</span>" +
                "<br>ESR:" + (this.ContainsFeature(this.Data, "ESR") ? "<span style=\"color:#13224D;font-family:Segoe UI;\">YES" : "<span style=\"color:#D71B0E;\">NO") + "</span>" + "</span>";

            this.element3.Text = "<html><span style=\"color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;\">" +
                "MP3 Player:" + (this.ContainsFeature(this.Data, "Mp3Player") ? "<span style=\"color:#13224D;font-family:Segoe UI;\">YES" : "<span style=\"color:#D71B0E;\">NO") + "</span>" +
                "<br>DVD Player:" + (this.ContainsFeature(this.Data, "DVDPlayer") ? "<span style=\"color:#13224D;font-family:Segoe UI;\">YES" : "<span style=\"color:#D71B0E;\">NO") + "</span>" + "</span>";

            this.element4.Text = "<html><span style=\"color:#010102;font-size:10.5pt;font-family:Segoe UI Semibold;\">" +
                "Air Conditioner:" + (this.ContainsFeature(this.Data, "AirConditioner") ? "<span style=\"color:#13224D;font-family:Segoe UI;\">YES" : "<span style=\"color:#D71B0E;\">NO") + "</span>" +
                "<br>Navigation:" + (this.ContainsFeature(this.Data, "Navigation") ? "<span style=\"color:#13224D;font-family:Segoe UI;\">YES" : "<span style=\"color:#D71B0E;\">NO") + "</span>" + "</span>";

            this.TextAlignment = ContentAlignment.TopLeft;
        }

        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(SimpleListViewVisualItem);
            }
        }
    }
}
